csrc/root_domain_map.h

```diff
@@ -438,58 +438,64 @@ class ComputeAtRootDomainMapBuilder : private BackwardVisitor {
 
   //! Map pointwise IterDomains from inputs of expressions to outputs.
   //! Do not map reduction IterDomains in inputs.
-  void mapPointwiseOrReductionOp(Expr* e);
+  void mapPointwiseLikeOp(Expr* e);
 
   using BackwardVisitor::handle;
 
   void dispatch(Expr* e) override;
 
   void handle(UnaryOp* uop) override {
-    mapPointwiseOrReductionOp(uop);
+    mapPointwiseLikeOp(uop);
   }
 
   void handle(BinaryOp* bop) override {
-    mapPointwiseOrReductionOp(bop);
+    mapPointwiseLikeOp(bop);
   }
 
   void handle(TernaryOp* top) override {
-    mapPointwiseOrReductionOp(top);
+    mapPointwiseLikeOp(top);
   }
 
   void handle(RNGOp* top) override;
 
-  void handle(IndexSelectOp* top) override {
-    mapPointwiseOrReductionOp(top);
```

Comment:
This wasn't correct before as the previous version of `mapPointwiseOrReductionOp` did not consider indexed IDs, which are now properly handled by `PairwiseRootDomainMap`.

Reply:
Can we also use this `mapPointwiseLikeOp` for broadcast and squeeze? Looks like every op should be handled by this function.

Reply:
I think we will need to update `new_broadcast_domains_` and `removed_broadcast_domains_` besides doing `mapPointwiseLikeOp`

Reply:
Yes, the broadcast analysis also needs to be done.

I didn't want to spend too much effort to clean up this class as this should eventually be replaced with IDGraph. The cleanup I did in this PR was part of the fix for #961.